Deprecate devServer.reactHot()

1. There are several open issues and PRs: #56, #131.
2. It’s not generic, works only for React.
3. It’s only one of possible hot reload solutions for React.
4. None of them are actively maintained.
5. The need for it is overvalued in my opinion, default webpack hot reload is good enough.
6. **It was always broken in 1.0**

```diff
// test-app/webpack.config.babel.js
 const developmentConfig = () => group([
   sourceMaps(),
   devServer(),
+  devServer.reactHot(),
   devServer.proxy({
     '/api/*': { target: 'http://localhost:4000' }
   }),
```

→

```
ERROR in   Error: Child compilation failed:
  Module not found: Error: Can't resolve 'react-hot' in '/Users/sapegin/webpack-blocks/test-app'
  BREAKING CHANGE: It's no longer allowed to omit the '-loader' suffix when using loaders.
                   You need to specify 'react-hot-loader' instead of 'react-hot',
                   see https://webpack.js.org/guides/migrating/#automatic-loader-module-name-extension-removed:
```

--- a/packages/dev-server/CHANGELOG.md
+++ b/packages/dev-server/CHANGELOG.md
@@ -1,5 +1,9 @@
 # @webpack-blocks/dev-server - Changelog
 
+## Next version
+
+- Removed `reactHot` ([#214](https://github.com/andywer/webpack-blocks/pull/214))
+
 ## 1.0.0-beta.3
 
 - Update `webpack-dev-server` to v2.6.1, now compatible with webpack v3 ([#179](https://github.com/andywer/webpack-blocks/issues/179))

--- a/packages/dev-server/README.md
+++ b/packages/dev-server/README.md
@@ -24,9 +24,6 @@ module.exports = createConfig([
     devServer.proxy({
       '/api': { target: 'http://localhost:3000' }
     }),
-    // in case you are using react-hot-loader 2.x:
-    // (need to install `react-hot-loader` package manually, though)
-    devServer.reactHot()
   ])
 ])
 ```

--- a/packages/dev-server/index.js
+++ b/packages/dev-server/index.js
@@ -7,7 +7,6 @@
 module.exports = devServer
 
 devServer.proxy = proxy
-devServer.reactHot = reactHot
 
 /**
  * @param {object} [options]    See https://webpack.github.io/docs/configuration.html#devserver
@@ -85,19 +84,4 @@ function proxy (proxyRoutes) {
       proxy: proxyRoutes
     }
   })
-}
-
-/**
- * For adding the react-hot-loader to the JS loaders.
- * @param {object} [options]
- * @param {RegExp, Function, string}  [options.exclude]   Directories to exclude.
- * @return {Function}
- */
-function reactHot () {
-  return (context, util) => util.addLoader(
-    Object.assign({
-      test: /\.(js|jsx)$/,
-      use: [ 'react-hot' ]
-    }, context.match)
-  )
 }

